File: credit.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    //Gets the Card Number from the user..
    long cardnumber = get_long("Number : ");

    long temp = cardnumber;
    int size = 0, sum = 0, x = 1, hundredsplace;

    while (temp != 0)
    {
        //To find the length of the Card Number and the First Two Digits.
        temp /= 10;
        size++;

        if (temp < 100 && temp > 9)
        {
            hundredsplace = temp;

        }
    }

    temp = cardnumber;

    if (size == 13 || size == 15 || size == 16)
    {

        //To check in which category the given card falls into.
        while (temp != 0)
        {
            if (x == 1)
            {
                sum += temp % 10;
                x = 0;

            }
            else
            {
                sum += (2 * (temp % 10) % 10) + (2 * (temp % 10) / 10);
                x = 1;
                
            }
            
            temp /= 10;

        }

        if (sum % 10 == 0)
        {
            //Checks if Card is a valid American Express Card
            if (size == 15 && (hundredsplace == 34 || hundredsplace == 37))
            {
                printf("AMERICAN EXPRESS\n");
                return 0;

            }
            //Checks if  Card is a valid Mastercard
            else if (size == 16 && (hundredsplace > 50 && hundredsplace < 56))
            {
                printf("MASTERCARD\n");
                return 0;

            }
            //Checks if  Card is a valid VISA Card
            else if ((size == 13 || size == 16) && hundredsplace / 10 == 4)
            {
                printf("VISA\n");
                return 0;

            }
        }
    }

    //Prints Invalid if card is not a Valid Credit Card.
    printf("INVALID\n");
    return 0;

}

File: credit.py
# Gets the Card Number from the user.
cardnumber = input("Number : ")

# To find the length of the Card Number and the First Two Digits.
size = len(cardnumber)
hundredsplace = 10 * int(cardnumber[0]) + int(cardnumber[1])
sum = 0

# Finding the sum using Luhn's credit card algorithm validator
for i in range(size):
    if i%2 == 0:
        sum += int(cardnumber[size - i - 1])
    else:
        temp = 2 * int(cardnumber[size - i - 1])
        sum += int(temp/10) + int(temp%10)

if size == 13 or size == 15 or size == 16:

    if sum % 10 == 0:
        
        # Checks if Card is a valid American Express Card.
        if size == 15 and (hundredsplace == 34 or hundredsplace == 37):
            print("AMEX")
            exit(0)
        
        # Checks if  Card is a valid Mastercard.
        elif size == 16 and hundredsplace > 50 and hundredsplace < 56:
            print("MASTERCARD")
            exit(0)
        
        # Checks if  Card is a valid VISA Card.
        elif (size == 13 or size == 16) and hundredsplace / 10 == 4:
            print("VISA")
            exit(0)

# Prints Invalid if card doesnot satisfy any of the above Conditions.
print("INVALID")
exit(0)